extensions/http-curl/client/HTTPClient.cpp

```diff
@@ -327,7 +356,9 @@ void HTTPClient::set_request_method(const std::string method) {
   if (my_method == "POST") {
     curl_easy_setopt(http_session_, CURLOPT_POST, 1L);
   } else if (my_method == "PUT") {
-    curl_easy_setopt(http_session_, CURLOPT_PUT, 0L);
+    curl_easy_setopt(http_session_, CURLOPT_UPLOAD, 1L);
```

Comment:
CURLOPT_PUT was set to 0 in this case, which means do not use PUT, but CURLOPT_PUT has been deprecated a long time ago anyway.